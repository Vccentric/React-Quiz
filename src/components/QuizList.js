import React from "react";
import PropTypes from "prop-types";
import QuizItem from "./QuizItem";

// component that contains a list of quizzes
const QuizList = ({ data, setQuiz }) => {
  // setup elements
  const quizzes = data?.quizzes?.map((quiz, index) => {
    const key = index * Math.random();
    return (
      <li key={key}>
        <QuizItem quiz={quiz} setQuiz={setQuiz} />
      </li>
    );
  });

  return (
    <div id="quiz-list">
      <div className="header">
        <h1 className="title">Welcome!!!</h1>
      </div>
      <div className="body">
        <h2 className="title">Quizzes:</h2>
        <ul>{quizzes}</ul>
      </div>
    </div>
  );
};

QuizList.propTypes = {
  data: PropTypes.object,
  setQuiz: PropTypes.func,
};

export default QuizList;
