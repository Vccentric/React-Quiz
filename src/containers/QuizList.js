import React from "react";
import PropTypes from "prop-types";
import Quiz from "../components/Quiz";

// container component that contains a list of quizzes
const QuizList = ({ data }) => {
  const quizzes = data?.quizzes?.map((quiz, index) => {
    const key = index * Math.random();
    const name = quiz.name || "";
    const total = quiz.questions?.length || 0;
    return (
      <li key={key}>
        <Quiz name={name} total={total} />
      </li>
    );
  });
  return (
    <div id="quiz-list">
      <div id="header">
        <h1 className="title">Lets Play!!!</h1>
        <h2 className="sub-title">And be the first!</h2>
      </div>
      <div id="body">
        <h2 className="title">Quizzes:</h2>
        <ul>{quizzes}</ul>
      </div>
    </div>
  );
};

QuizList.propTypes = {
  data: PropTypes.object,
};

export default QuizList;
