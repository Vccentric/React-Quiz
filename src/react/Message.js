import React from 'react';
import { hot } from 'react-hot-loader';

// message component
function Message(props) {
    const { isAnswered, isCorrect } = props;
    return (
        <div
            id="message"
            style={isAnswered ? { display: 'block' } : { display: 'none' }}
            className={isCorrect ? 'correct' : 'wrong'}
        >
            {isCorrect ? 'Correct Answer!' : 'Wrong Answer!'}
        </div>
    );
}

export default hot(module)(Message);
