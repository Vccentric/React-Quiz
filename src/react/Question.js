import React from 'react';
import { hot } from 'react-hot-loader';

// question component
function Question(props) {
    const { question } = props;
    return <p id="question">{question}</p>;
}

export default hot(module)(Question);
