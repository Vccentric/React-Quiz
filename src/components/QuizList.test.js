import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import React from "react";
import QuizList from "./QuizList";

// setup variables
const setQuiz = jest.fn();

test("renders correctly - snapshot", () => {
  const tree = renderer
    .create(<QuizList data={{}} setQuiz={setQuiz} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders results with mock values", () => {
  const mockData = { quizzes: [{ name: "abc" }, { name: "def" }] };
  render(<QuizList data={mockData} setQuiz={setQuiz} />);
  const item1 = screen.getByText("abc");
  const item2 = screen.getByText("def");
  expect(item1).toBeInTheDocument();
  expect(item2).toBeInTheDocument();
});
