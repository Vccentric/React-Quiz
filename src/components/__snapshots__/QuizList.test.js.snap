// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - snapshot 1`] = `
<div
  id="quiz-list"
>
  <div
    className="header"
  >
    <h1
      className="title"
    >
      Welcome!!!
    </h1>
  </div>
  <div
    className="body"
  >
    <h2
      className="title"
    >
      Quizzes:
    </h2>
    <ul />
  </div>
</div>
`;
